src/gui/qgsmapcanvas.cpp

```diff
@@ -644,6 +675,17 @@ void QgsMapCanvas::refreshMap()
   stopPreviewJobs();
 
   mSettings.setExpressionContext( createExpressionContext() );
+
+  // if using the temporal controller in animation mode, get the frame settings from that
+  if ( QgsTemporalNavigationObject *to = dynamic_cast < QgsTemporalNavigationObject * >( mController ) )
```

Comment:
@nyalldawson , you've called previous pointers "no", while this one is "to". Could you harmonize and maybe use a slightly expanded name, like navObj or something? :)

Reply:
Done